File: BPM.Web.Distributor.UI/Models/AuthResponse.cs
using BPM.Web.Distributor.UI.Models;
using BPM.Web.Distributor.UI.Models.DTOs;

namespace BPM.Web.Distributor.UI.Models
{
    public class AuthResponse
    {
        public string Message { get; set; }

        public bool IsValidUser { get; set; }

        public bool IsValidPassword { get; set; }

        public string JwtToken { get; set; }

        public string RefreshToken { get; set; }

        public AuthenticateResponseDto AuthenticateResponseDto { get; set; }
    }
}

File: BPM.Web.Dealer.UI/src/app/models/auth-response.ts
import { isActive } from "@angular/router";

// Define the authentication response interface
export interface AuthResponse {
  authenticateResponseDto: {
    userId: string;
    firstName: string;
    lastName: string;
    email: string;
    phone: string;
    isActive: boolean;
    roleId: string;
    dealerId: string;
    dealerInfo: {
      id: string;
      dealershipName: string;
      contactPerson: string;
      email: string;
      phone: string;
      alternatePhone: string;
      addressLine1: string;
      addressLine2: string;
      city: string;
      state: string;
      country: string;
      postalCode: string;
      gstNumber: string;
      registrationNumber: string;
      tradeLicenseNumber: string;
      website: string;
      isActive: boolean;
    },
    roleInfo: {
      id: string,
      name: string,
      isActive: boolean,
      code: string
    }
  };
  jwtToken: string;
  refreshToken: string;
  message: string;
  isValidUser: boolean;
  isValidPassword: boolean;
}
